bootstrap/_deb_common.sh

```diff
@@ -10,26 +10,37 @@
 #     - 7.8 "wheezy" (x64)
 #     - 8.0 "jessie" (x64)
 
-# virtualenv binary can be found in different packages depending on
-# distro version (#346)
-distro=$(lsb_release -si)
-# 6.0.10 => 60, 14.04 => 1404
-version=$(lsb_release -sr | awk -F '.' '{print $1 $2}')
-if [ "$distro" = "Ubuntu" -a "$version" -ge 1410 ]
-then
-  virtualenv="virtualenv"
-elif [ "$distro" = "Debian" -a "$version" -ge 80 ]
-then
-  virtualenv="virtualenv"
-else
-  virtualenv="python-virtualenv"
-fi
+apt-get update
 
 # dpkg-dev: dpkg-architecture binary necessary to compile M2Crypto, c.f.
 #           #276, https://github.com/martinpaljak/M2Crypto/issues/62,
 #           M2Crypto setup.py:add_multiarch_paths
 
-apt-get update
-apt-get install -y --no-install-recommends \
-  python python-setuptools "$virtualenv" python-dev gcc swig \
-  dialog libaugeas0 libssl-dev libffi-dev ca-certificates dpkg-dev
+common_packages="python python-setuptools python-dev gcc swig dialog libaugeas0 libssl-dev openssl libffi-dev"
+
+if [ "$1" = "no_venv" ]
+then
+  packages="$common_packages"
+else
+
+  # virtualenv binary can be found in different packages depending on
+  # distro version (#346)
+  distro=$(lsb_release -si)
+  # 6.0.10 => 60, 14.04 => 1404
+  version=$(lsb_release -sr | awk -F '.' '{print $1 $2}')
+  if [ "$distro" = "Ubuntu" -a "$version" -ge 1410 ]
+  then
+    virtualenv="virtualenv"
+  elif [ "$distro" = "Debian" -a "$version" -ge 80 ]
+  then
+    virtualenv="virtualenv"
+  else
+    virtualenv="python-virtualenv"
+  fi
+
+  packages="$virtualenv $common_packages"
+fi
+
+apt-get install -y --no-install-recommends $packages
+
+ln -s /usr/include/x86_64-linux-gnu/openssl/opensslconf.h /usr/include/openssl/
```

Comment:
Building M2Crypto on Ubuntu 14.04 in both Docker and Vagrant fails without this symlink.

Error for Vagrant:

```
Installed /vagrant/venv/lib/python2.7/site-packages/letsencrypt-0.1-py2.7.egg
Processing dependencies for letsencrypt==0.1
Searching for M2Crypto
Reading https://pypi.python.org/simple/M2Crypto/
Best match: M2Crypto 0.22.3
Downloading https://pypi.python.org/packages/source/M/M2Crypto/M2Crypto-0.22.3.tar.gz#md5=573f21aaac7d5c9549798e72ffcefedd
Processing M2Crypto-0.22.3.tar.gz
Writing /tmp/easy_install-fXvJeI/M2Crypto-0.22.3/setup.cfg
Running M2Crypto-0.22.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-fXvJeI/M2Crypto-0.22.3/egg-dist-tmp-blEg2E


Stderr from the command:

stdin: is not a tty
dpkg-preconfigure: unable to re-open stdin: No such file or directory
SWIG/_evp.i:12: Error: Unable to find 'openssl/opensslconf.h'
SWIG/_ec.i:7: Error: Unable to find 'openssl/opensslconf.h'
error: Setup script exited with error: command 'swig' failed with exit status 1
```

Error for Docker:

```
Downloading https://pypi.python.org/packages/source/M/M2Crypto/M2Crypto-0.22.3.tar.gz#md5=573f21aaac7d5c9549798e72ffcefedd
Processing M2Crypto-0.22.3.tar.gz
Writing /tmp/easy_install-St43Ji/M2Crypto-0.22.3/setup.cfg
Running M2Crypto-0.22.3/setup.py -q bdist_egg --dist-dir /tmp/easy_install-St43Ji/M2Crypto-0.22.3/egg-dist-tmp-iRzIHw
SWIG/_evp.i:12: Error: Unable to find 'openssl/opensslconf.h'
SWIG/_ec.i:7: Error: Unable to find 'openssl/opensslconf.h'
error: Setup script exited with error: command 'swig' failed with exit status 1
```